src/xenusbdevice/UsbConfig.cpp

```diff
@@ -1784,75 +1784,30 @@ GetString(
     AcquireFdoLock(fdoContext);
     do
     {
-
-        USHORT ActualLength;
-
-        NTSTATUS Status = GetDescriptor(
-                              fdoContext,
-                              &fdoContext->ScratchPad.Packet,
-                              USB_STRING_DESCRIPTOR_TYPE,
-                              BMREQUEST_TO_DEVICE,
-                              index,
-                              fdoContext->LangId,
-                              4,
-                              sizeof(USB_STRING));
-
-        if (!NT_SUCCESS(Status))
-        {
-            TraceEvents(TRACE_LEVEL_ERROR, TRACE_DEVICE,
-                        __FUNCTION__": GetDescriptor length probe failed %x\n",
-                        Status);
-            ExFreePool(uString);
-            uString = NULL;
-            break;
-        }
-
-        ActualLength = (USHORT) ((PUCHAR)fdoContext->ScratchPad.Buffer)[0];
+        // Default to a read of 0xff since some devices break from the probe followed by another read and
+        // can lead to DPC and xen event channel being torn down
         TraceEvents(TRACE_LEVEL_INFORMATION, TRACE_DEVICE,
-                    __FUNCTION__": GetDescriptor requesting string of length %d (0x%x)\n",
-                    ActualLength, ActualLength);
+                    __FUNCTION__": GetDescriptor requesting string of length 255 (0xff)\n");
 
-        Status = GetDescriptor(
+        NTSTATUS Status = GetDescriptor(
                      fdoContext,
                      &fdoContext->ScratchPad.Packet,
                      USB_STRING_DESCRIPTOR_TYPE,
                      BMREQUEST_TO_DEVICE,
                      index,
                      fdoContext->LangId,
-                     ActualLength,
+                     0xff,
                      sizeof(USB_STRING));
 
         if (!NT_SUCCESS(Status))
         {
-            TraceEvents(TRACE_LEVEL_WARNING, TRACE_DEVICE,
-                        __FUNCTION__": %s actual length %d GetDescriptor failed %x\n",
+            TraceEvents(TRACE_LEVEL_ERROR, TRACE_DEVICE,
+                        __FUNCTION__": %s string length 255 GetDescriptor failed %x\n",
                         fdoContext->FrontEndPath,
-                        ActualLength,
                         Status);
-            //
-            // ugh. Ok try using 255. Some devices are broken.
-            // Note: some other devices are broken the other way.
```

Comment:
This comment says some devices don't like just using 255 :(  It was in the initial open sourcing, so there is no further detail.

Reply:
Nice debugging, @Burrch3s.  I followed the linux code a little deeper and found the 255 value in here:
https://elixir.bootlin.com/linux/v5.4.140/source/drivers/usb/core/message.c#L729

So it looks Linux does the opposite of this driver - 255 first with a fall back to length + string.  What do you think about making that change?

I'm concerned if you've only tested on 2 devices that there may be a regression by switching to only the 255 read.  If you match the linux behavior, then maybe that would be the most compatible option?